### PR TITLE
samples: sensor: Add sample for Si7006

### PR DESCRIPTION
**Pull request by thorvald**
This also updates the documentation to relfect the same driver works
for any of Si7006/Si7013/Si7020/Si7021, and adds the definition to the
SLWSTK6121A and STK3701A boards.
Sample code was based on sample code for BME280 sensor.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

### CODE REVIEW

File: drivers/i2c/i2c_gecko.c [hunk 1]
```diff
@@ -262,3 +262,36 @@ DEVICE_AND_API_INIT(i2c_gecko_1, DT_INST_LABEL(1),
 		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &i2c_gecko_driver_api);
 #endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(2), okay)
```
Comment (MaureenHelm) [inline drivers/i2c/i2c_gecko.c:266]: Can you look at updating the driver to use an instance macro and DT_INST_FOREACH_STATUS_OKAY?

Comment (thorvald) [reply drivers/i2c/i2c_gecko.c]: Done. This looks so much nicer :)

I don't have any kits with a SOC old enough to test the path without ```CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION```. Also, it looks like none of the boards that use such a SOC have any i2c nodes defined. So testing for that code-path is limited to modifying the DTS for a matching board to include a i2c definition, and ensuring that this compiles for that board, and that it doesn't compile if the location definitions don't match.

**Review by chrta — APPROVED** [review]
Tested sample successfully on efm32gg_stk3701a

File: samples/sensor/si7006/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+sample:
+  name: Si7006 Sensor sample
+tests:
+  sample.sensor.si7006:
+    build_only: true
+    platform_allow: efm32gg11_wlstk6121a
```
Comment (MaureenHelm) [inline samples/sensor/si7006/sample.yaml:6]: How about instead:
```
    filter: dt_compat_enabled("silabs,si7006")
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
one nit and needs a rebase, but otherwise looks good

**Pull request closed without merging**